test/sink/sink_service_test.cc

```diff
@@ -0,0 +1,305 @@
+#include <grpc++/grpc++.h>
+
+#include <vector>
+
+#include "external/envoy/test/test_common/environment.h"
+#include "external/envoy/test/test_common/network_utility.h"
+#include "external/envoy/test/test_common/utility.h"
+
+#include "api/sink/sink.pb.h"
+
+#include "sink/service_impl.h"
+
+#include "test/mocks/sink/mock_sink.h"
+
+#include "absl/synchronization/notification.h"
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+namespace {
+
+using ::nighthawk::SinkRequest;
+using ::nighthawk::SinkResponse;
+using ::nighthawk::StoreExecutionRequest;
+using ::nighthawk::StoreExecutionResponse;
+using ::nighthawk::client::ExecutionResponse;
+using ::testing::HasSubstr;
+using ::testing::Return;
+using ::testing::TestWithParam;
+using ::testing::ValuesIn;
+
+class SinkServiceTest : public TestWithParam<Envoy::Network::Address::IpVersion> {
+public:
+  void SetUp() override {
+    auto sink = std::make_unique<MockSink>();
+    sink_ = sink.get();
+    service_ = std::make_unique<SinkServiceImpl>(std::move(sink));
+    grpc::ServerBuilder builder;
+    loopback_address_ = Envoy::Network::Test::getLoopbackAddressUrlString(GetParam());
+
+    builder.AddListeningPort(fmt::format("{}:0", loopback_address_),
+                             grpc::InsecureServerCredentials(), &grpc_server_port_);
+    builder.RegisterService(service_.get());
+    server_ = builder.BuildAndStart();
+    setupGrpcClient();
+  }
+
+  void TearDown() override { server_->Shutdown(); }
+
+  void setupGrpcClient() {
+    channel_ = grpc::CreateChannel(fmt::format("{}:{}", loopback_address_, grpc_server_port_),
+                                   grpc::InsecureChannelCredentials());
+    stub_ = std::make_unique<nighthawk::NighthawkSink::Stub>(channel_);
+  }
+
+  MockSink* sink_;
+  std::unique_ptr<SinkServiceImpl> service_;
+  std::unique_ptr<grpc::Server> server_;
+  std::shared_ptr<grpc::Channel> channel_;
+  grpc::ClientContext context_;
+  SinkRequest request_;
+  SinkResponse response_;
+  std::unique_ptr<nighthawk::NighthawkSink::Stub> stub_;
+  std::string loopback_address_;
+  int grpc_server_port_{0};
+};
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, SinkServiceTest,
+                         ValuesIn(Envoy::TestEnvironment::getIpVersionsForTest()),
+                         Envoy::TestUtility::ipTestParamsToString);
+
+TEST_P(SinkServiceTest, LoadSingleResultWithJustExecutionResponse) {
+  // Our mock sink will yield a single result, with the correct execution id.
+  const std::string kTestId = "test-id";
+  absl::StatusOr<std::vector<ExecutionResponse>> response_from_mock_sink =
+      std::vector<ExecutionResponse>{{}};
+  ExecutionResponse& response = response_from_mock_sink.value().at(0);
+  response.mutable_execution_id()->assign(kTestId);
```

Comment:
`.assign()` accepts the `abs::string_view` we're passing it, with `.set_execution_id` we will have to create a `std::string` because passing the string_view directly will not work. Personally I find this slightly nicer to look at, but if you feel strongly about it I can change this here (in this PR, but there are some other spots where this is done too).

Reply:
I see, I was afraid it would be `string_view` not working. No problem keeping `.assign()`

Reply:
i don't have a problem here, but we aren't passing a string_view here, are we?  we're just passing a const string.

Reply:
Oh yes, changed this in all the appropriate spots in 8492073